src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,24 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super(popupSelector);
+    this._confirmDelete = this._popupElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._confirmDelete.addEventListener("submit", (event) => {
+      event.preventDefault();
+      this._handleFormSubmit();
+      this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and send it again.